openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -264,30 +269,38 @@ class CourseWaffleFlag(WaffleFlag):
     Uses a cached waffle namespace.
     """
 
-    def _get_course_override_callback(self, course_id):
+    def _get_course_override_callback(self, course_key):
         """
         Returns a function to use as the check_before_waffle_callback.
 
         Arguments:
-            course_id (CourseKey): The course to check for override before
+            course_key (CourseKey): The course to check for override before
             checking waffle.
         """
         def course_override_callback(namespaced_flag_name):
             """
             Returns True/False if the flag was forced on or off for the provided
             course.  Returns None if the flag was not overridden.
 
+            Note: Has side effect of caching the override value.
+
             Arguments:
                 namespaced_flag_name (String): A namespaced version of the flag
                     to check.
             """
-            force_override = WaffleFlagCourseOverrideModel.override_value(namespaced_flag_name, course_id)
+            cache_key = '{}.{}'.format(namespaced_flag_name, course_key.to_deprecated_string())
```

Comment:
The suggested form is `unicode(course_key)` not `course_key.to_deprecated_string()`.